init.sls

```diff
@@ -110,7 +112,11 @@ chown_pgdata:
 createuser-{{ index }}:
   postgres_user.present:
     - name: {{ config['username'] }}
+    {% if config['method'] == 'scram-sha-256' %}
+    - encrypted: scram-sha-256
```

Comment:
According to the postgres docs, `scram-sha-256` is supported since at least pg-10. As it is the more secure option, I think it would make a good default.

Using `md5` (which we can set explicitly instead of `True` I think) is only necessary for older clients, which I don't think are used that much. 

I'd suggest defaulting to `scram-sha-256` unless there's a `method` explicitly set.

wdyt?

Reply:
Also ok to default to `md5` and change the default in another pull request if you prefer.

Reply:
I would agree the defaulting to `scram-sha-256` is a good think. 

I just wanted to make sure that it not break some thinks. Not sure what will happen if you have active connections with `md5` and the user will be recreated with `scram-sha-256` and the `pg_hba.conf` file be changed.

Reply:
If you don't think this is a risk. Then I can also change the default.

Reply:
I'll bump the version to indicate changed behavior. `method: md5` is the only thing required to return to the old behavior, I think that's alright.

Reply:
Make sure to add the option to `pillar.example`, though.

Reply:
Pushed an update.